src/lib/y2storage/encryption.rb

```diff
@@ -188,6 +188,22 @@ def method
     # @param value [Encryption::Processes]
     def encryption_process=(value)
       save_userdata(:encryption_process, value)
+      @encryption_process = value
```

Comment:
Because we modify the state of the `EncryptionProcess` data. If we wouldn't use an instance variable, every time we call `#encryption_process` we would be accessing the object that was stored in the devicegraph, with its former state. Thus, all the modifications would be lost.

That's also why we need `#save_encryption_process`, to persist the state.

Think about the userdata value as it was a database, every time you call `#userdata_value` you are reading the version of the object that was saved in the database, not the current version with the latest changes.